Source/engine/render/scrollrt.cpp

```diff
@@ -832,7 +836,7 @@ void DrawDungeon(const Surface &out, Point tilePosition, Point targetBufferPosit
 		if (tilePosition.x > 0 && tilePosition.y > 0 && targetBufferPosition.y > TILE_HEIGHT) {
 			char bArch = dSpecial[tilePosition.x - 1][tilePosition.y - 1];
 			if (bArch != 0) {
-				ClxDraw(out, targetBufferPosition + Displacement { 0, -TILE_HEIGHT }, (*pSpecialCels)[bArch - 1]);
+				ClxDraw(out, targetBufferPosition + Displacement { 0, -TILE_HEIGHT }, (*pSpecialCels)[--bArch]);

```

Comment:
why would you do this? Modifying bArch is pointless, we only want the value -1 ;)

Reply:
Compiler angry about potential OOB from sub expression bArch - 1

Reply:
Shouldn't you check if bArch > 0?

Reply:
Actually I think I might be able to just static cast the sub expression like I did in the other PR to resolve the compiler warning since it would be impossible to have an OOB index by the compiler's standards

Reply:
Not sure why static casting is better here. If you don't need negatives, make it unsigned. If you do, make sure it's greater than zero. If you static cast a negative, it's just going to be a big number that might be OOB on pSpecialCels anyway.

Reply:
I would prefer that it be modified on it's own line right after checking it if that is needed.